tracee-ebpf/tracee/tracee.bpf.c

```diff
@@ -2545,6 +2552,77 @@ int tracepoint__sched__sched_switch(struct bpf_raw_tracepoint_args *ctx)
     return events_perf_submit(&data, SCHED_SWITCH, 0);
 }
 
+SEC("kprobe/security_file_ioctl")
+int BPF_KPROBE(trace_security_file_ioctl)
+{
+    event_data_t data = {};
+    if (!init_event_data(&data, ctx))
+        return 0;
+    if (get_config(CONFIG_TRACEE_PID) != data.context.host_pid)
+        return 0;
+    int cmd = PT_REGS_PARM2(ctx);
+
+    if (cmd != 65)
+        return 0;
+
+    int key =0;
+    u64* syscall_tableP = bpf_map_lookup_elem(&syscall_table_addr_map, (void *)&key);
+    u64* stextP = bpf_map_lookup_elem(&code_boundaries_map, (void *)&key);
+    key++;
+    u64 * etextP = bpf_map_lookup_elem(&code_boundaries_map, (void *)&key);
+
+    if (syscall_tableP == NULL)
+        return 0;
+    if (stextP == NULL)
+        return 0;
+    if (etextP == NULL)
+        return 0;
+    unsigned long *syscall_table_addr = (unsigned long*)*syscall_tableP;
+    unsigned long stext = *stextP;
+    unsigned long etext = *etextP;
+    unsigned long syscall_addr =0;
+
+    // checking the syscalls:
+    // read, write, open, close
+    // kill, ioctl, getdents, getdents64,socket
+    // execve, execveat , ptrace ,bpf
+
+
+    //list of all sysaclls number we want to check
+//	int syscalls_to_check[13]={0};
+#if defined(bpf_target_x86)
+    int syscalls_to_check[] = {0, 1, 2, 3, 16, 62, 78, 217, 41, 59, 322, 101, 321};
+#elif defined(bpf_target_arm64)
+     int syscalls_to_check[] = {3, 4, 5, 6, 11, 26, 37, 141, 217, 281, 322, 386, 387};
+#endif
+    //check if the array is initialized
+//    if (syscalls_to_check[2] ==0)
+//        return 0;
+
+
+
+    int i=0 ,syscall_number;
+    #pragma unroll
+    for(int syscall_index=0;syscall_index<10;syscall_index++)
```

Comment:
This is problematic because multiple arches might have different amount of syscalls to be checked. The aarch64 syscall numbers were wrong (please double check). Some syscalls didn't exist for aarch64.

In tracee.bpf.c I suggested a sizeof(array of ints)/4, each syscall number in a different line (to help code sustainability). 

But in golang side, we have to replicate this array:

https://github.com/rafaeldtinoco/tracee/commit/a499519e210b9bd123fa988cd28040f31be2528b#diff-4aea1bf5a79623d508c2193216f7dce24e18faed63b9a947d177a8cd787eabf6R2629

with a similar #ifdef (per arch) and use the sizeof this array (instead of using a const value).